src/snap/metric.h

```diff
@@ -25,35 +25,140 @@ limitations under the License.
 #include "snap/config.h"
 
 namespace Plugin {
+   
+    class NamespaceElement{
+        public:
+
+        /** 
+        * Constructor which makes "NamespaceElement" from three strings:
+        * value,
+        * name (optional- default parameter is empty),
+        * description (optional- default parameter is empty).
+        */
+        NamespaceElement(std::string val, std::string nam ="", std::string desc=""); 
+
+        /**
+        * Default empty constructor.
+        */        
+        NamespaceElement();
+
+        /**
+        * Default empty destructor.
+        */
+        ~NamespaceElement();
+
+        /**
+        * Setters for value, name and description
+        */
+        void set_value(std::string v);
+        void set_name(std::string n);
+        void set_description(std::string d);   
+
+        /**
+        * Getters for value, name and description
+        */
+        const std::string get_value() const;
+        const std::string get_name() const;
+        const std::string get_description() const;
+
+        /**
+        * is_dynamic returns true if "NamespaceElement" has noempty name field.
```

Comment:
As we want to have parity for snap-plugin-lib-go, could your provide similar description to this one - https://github.com/intelsdi-x/snap-plugin-lib-go/blob/master/v1/plugin/metric.go#L328 . In my opinion it is a little bit clearer.